lib/fog/aws/models/compute/volume.rb

```diff
@@ -109,22 +97,15 @@ def force_detach
           detach(true)
         end
 
-        private
-
-        def attachmentSet=(new_attachment_set)
-          merge_attributes(new_attachment_set.first || {})
-        end
-
-        def attach(new_server)
+        def attach(new_server, new_device)
```

Comment:
Could you elaborate on why the new argument is needed? The rest looks good, but I don't think I'm quite groking this. Thanks!

Reply:
Tests asserted this procedure

* `v = service.volumes.create(device: '/dev/sda1', **)`
* `v.server = service.instances.create(**) # requires :device`

In step 1, a device is provided to a volume instance. In step 2, the model calls `Fog.wait_for { ready? }` (implicit reload) and then attaches the server.

The issue is between step 1 and 2, there is a property of the local model (`device = '/dev/sda1'`) that does not match the remote model (`device = nil`).  

On `reload`: 

* `fog-core 1.x`, the `device` attribute retained it's value.  
* `fog-core 2.x`, `device` is set the correct remote value (`nil`).

IMO a `reload` should put the local model in the same state as the remote model.  To enforce that, I've removed the ability for `volume.server = server` to attach volume and instead require calling `volume.attach(server, device)`.